src/components/SurahInfo/index.js

```diff
@@ -0,0 +1,84 @@
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+const style = require('./style.scss');
+
+@connect(
+  (state, ownProps) => {
+    return {
+      isShowingInfo: state.surahs.isShowingInfo,
+      currentSurah: state.surahs.entities[state.surahs.current]
+    };
+  }
+)
+export default class SurahInfo extends Component {
+  constructor() {
+    super();
+
+    this.state = {
+      page: null
+    };
+  }
+
+  componentWillUpdate(nextProps, nextState) {
+    const { currentSurah } = this.props;
+
+    if (!nextProps.isShowingInfo) {
+      return;
+    }
+
+    const surahInfo = require(`./htmls/${currentSurah.id}.html.js`);
```

Comment:
this could be done in the render method. Don't need it here. And don't need to setState either cause that will run the component updates again.